File: week6/kotlin/adapter_pattern/Adapter.kt
// 외부 클래스
class ExternalClass {
    fun search(): String {
        return "get user info"
    }
}

// 내부 클래스 (인터페이스)
interface InternalClass {
    fun fetch(): String
}

// 어댑터 클래스
class Adapter(private val external: ExternalClass) : InternalClass {
    override fun fetch(): String {
        return external.search()
    }
}

fun main() {
    val external = ExternalClass()
    val adapter = Adapter(external)

    println(adapter.fetch()) // "get user info"
}

File: week6/cpp/adapter_pattern/Adapter.cpp
#include "Adapter.hpp"

// 외부 클래스의 메서드 구현
std::string ExternalClass::search() {
    return "get user info";
}

// 어댑터 클래스의 생성자 구현
Adapter::Adapter(ExternalClass& external) : external(external) {}

// 어댑터의 fetch 메서드 구현
std::string Adapter::fetch() {
    return external.search();
}
